### PR TITLE
docs(protocol): update mainnet deployment docs

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

**Review by Brechtpd — APPROVED** [review]

File: packages/protocol/deployments/mainnet-contract-logs-L1.md [hunk 1]
```diff
@@ -65,8 +65,7 @@
   - deployed on May 1, 2024 @commit`56dddf2b6`
   - admin.taiko.eth accepted the ownership @tx`0x0ed114fee6de4e3e2206cea44e6632ec0c4588f73648d98d8df5dc0183b07885`
   - upgraded from `0x91d593d34f2E1904cDCe3D5290a74563F87bCF6f` to `0x4A1091c2fb37D9C4a661c2384Ff539d94CCF853D` @commit`b90b932` @tx`0x416560cd96dc75ccffebe889e8d1ab3e08b33f814dc4a2bf7c6f9555071d1f6f`
-- todo:
-  - upgrade contract impl to enable selfeDelegate and remove unused code.
+  - upgraded from `0x4A1091c2fb37D9C4a661c2384Ff539d94CCF853D` to `0xc71CC3B0a47149878fad337fb2ca54E546A645ba` @commit`b955e0e` @tx`0x5a60c5815947a199cc84e1bc75539e01a202597b20c1f87bd9d02f8be6453abd`
```
Comment (dantaik) [inline packages/protocol/deployments/mainnet-contract-logs-L1.md:68]: change the `impl`  value above

File: packages/protocol/deployments/mainnet-contract-logs-L1.md [hunk 2]
```diff
@@ -212,9 +212,8 @@
   - deployed on May 1, 2024 @commit`56dddf2b6`
   - admin.taiko.eth accepted the ownership @tx`0x0ed114fee6de4e3e2206cea44e6632ec0c4588f73648d98d8df5dc0183b07885`
   - Upgraded from `0x717DC5E3814591790BcB1fD9259eEdA7c14ce9CF` to `0x750221E951b77a2Cb4046De41Ec5F6d1aa7942D2` @commit`b90b932` @tx`0x416560cd96dc75ccffebe889e8d1ab3e08b33f814dc4a2bf7c6f9555071d1f6f`
-  - Upgraded from `0x750221E951b77a2Cb4046De41Ec5F6d1aa7942D2` to `0x253E47F2b1e91F2001d3578aeB24C0ccF464b65es` @commit`cd5144255` @tx`0x8030569e293baddbc4e8b26688a1ecf14a231d86c90e9d02dad1e919ea2f3964`
-- todo:
-  - upgrade impl
+  - Upgraded from `0x750221E951b77a2Cb4046De41Ec5F6d1aa7942D2` to `0x253E47F2b1e91F2001d3578aeB24C0ccF464b65e` @commit`cd5144255` @tx`0x8030569e293baddbc4e8b26688a1ecf14a231d86c90e9d02dad1e919ea2f3964`
```
Comment (dantaik) [inline packages/protocol/deployments/mainnet-contract-logs-L1.md:215]: also change the `impl' field

File: packages/protocol/deployments/mainnet-contract-logs-L1.md [hunk 3]
```diff
@@ -235,8 +234,7 @@
   - deployed on May 1, 2024 @commit`56dddf2b6`
   - Upgraded from `0x717DC5E3814591790BcB1fD9259eEdA7c14ce9CF` to `0x750221E951b77a2Cb4046De41Ec5F6d1aa7942D2` @commit`b90b932` @tx`0x416560cd96dc75ccffebe889e8d1ab3e08b33f814dc4a2bf7c6f9555071d1f6f`
   - Upgraded from `0x750221E951b77a2Cb4046De41Ec5F6d1aa7942D2` to `0x253E47F2b1e91F2001d3578aeB24C0ccF464b65e` @commit`cd5144255` @tx`0x8030569e293baddbc4e8b26688a1ecf14a231d86c90e9d02dad1e919ea2f3964`
-- todo:
-  - upgrade impl
+  - Upgraded from `0x253E47F2b1e91F2001d3578aeB24C0ccF464b65e` to `0x468F6A9C0ad2e9C8370687D2844A9e70fE942d5c` @commit`b955e0e` @tx`0x5a60c5815947a199cc84e1bc75539e01a202597b20c1f87bd9d02f8be6453abd`
```
Comment (dantaik) [inline packages/protocol/deployments/mainnet-contract-logs-L1.md:237]: also change the `impl' field

File: packages/protocol/deployments/mainnet-contract-logs-L1.md (hunk 2 above)
Comment (dantaik) [inline packages/protocol/deployments/mainnet-contract-logs-L1.md:215]: 0x579A8d63a2Db646284CBFE31FE5082c9989E985c
not 0x253E47F2b1e91F2001d3578aeB24C0ccF464b65e

File: packages/protocol/deployments/mainnet-contract-logs-L1.md (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/deployments/mainnet-contract-logs-L1.md:68]: updated

File: packages/protocol/deployments/mainnet-contract-logs-L1.md (hunk 2 above)
Comment (davidtaikocha) [reply packages/protocol/deployments/mainnet-contract-logs-L1.md:215]: updated

Comment (dantaik) [inline packages/protocol/deployments/mainnet-contract-logs-L1.md:215]:
```suggestion
  - Upgraded from `0x253E47F2b1e91F2001d3578aeB24C0ccF464b65e` to `0x468F6A9C0ad2e9C8370687D2844A9e70fE942d5c` @commit`cd5144255` @tx`0x8030569e293baddbc4e8b26688a1ecf14a231d86c90e9d02dad1e919ea2f3964`
```

File: packages/protocol/deployments/mainnet-contract-logs-L1.md (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/deployments/mainnet-contract-logs-L1.md:68]: updated

File: packages/protocol/deployments/mainnet-contract-logs-L1.md (hunk 3 above)
Comment (davidtaikocha) [reply packages/protocol/deployments/mainnet-contract-logs-L1.md:237]: updated

**Review by dantaik — APPROVED** [review]

**Pull request merged**